packages/babel-plugin-transform-classes/src/vanilla.js

```diff
@@ -163,10 +163,27 @@ export default class ClassTransformer {
     //
     body.push(t.returnStatement(t.cloneNode(this.classRef)));
 
+    const strictParent = this.path.findParent(path => {
```

Comment:
looks better now 👍 would be great to add tests to cover this

also would be great to skip adding those directives in modules, but im not entirely sure how to check that, maybe you can check some property on the program or file? there are things like `sourceType` which could hint you this info, im not sure though where this info resides

Reply:
I'm not sure this is actually testable. Having the directive in the program or in a parent function produces the same result.

Reply:
You can test by writing a random non-simple arguments list (destructuring, defaults) in the IIFE the transform generates and seeing if it compiles.

`'use strict'` inside a function is too late to disable sloppy mode in the arguments list.

Reply:
Not sure if we have understood each other - I was asking if there is a possibility to skip adding this for such input
```js
```
import a from './a'
class B {
  constructor() { this.c = a() }
}
```

Modules are already strict be definition and they do not have to add use strict directive explicitly.

Reply:
Agreed. We'll want to make sure to avoid adding it to module. It should be safe to do 

```
if (path.isProgram() && path.node.sourceType === "module") {
  return true;
}
```
to this to cover the module case.